src/input_source.cpp

```diff
@@ -88,7 +88,7 @@ bool Input::Source::InitRecording(const std::string& record_to_path) {
 	if (!record_to_path.empty()) {
 		auto path = record_to_path.c_str();
 
-		record_log = FileFinder::OpenOutputStream(path, std::ios::out | std::ios::trunc);
+		record_log = std::make_unique<Filesystem_Stream::OutputStream>(FileFinder::OpenOutputStream(path, std::ios::out | std::ios::trunc));
```

Comment:
This doesn't make sense, you create an `OutputStream` temporary, and then move it into a unique_ptr.

Now that your `OutputStream` itself is movable we don't need the `unique_ptr` at all.

Just do 

`auto record_log = FileFinder::OpenOutputStream(path, std::ios::out | std::ios::trunc);`

If someone wants to use the stream:
````
void use(OutputStream& s);
use(record_log)
```

If someone want so take ownership
```
take(OutputStream s)
take(std::move(s));
```